corehq/apps/app_manager/templates/app_manager/partials/case_list_properties.html

```diff
@@ -18,11 +18,13 @@
                 <td class="detail-screen-icon" data-bind="jqueryElement: $grip, visible: $root.edit"></td>
                 <!--ko if: !isTab -->
                     <td class="detail-screen-field control-group"
-                        data-bind="css: {error: field.value && !DetailScreenConfig.field_val_re.test(field.value)}">
+                        data-bind="css: {
+                            error: field.observableVal() && !DetailScreenConfig.field_val_re.test(field.observableVal())
+                        }">
                         <div data-bind="html: CC_DETAIL_SCREEN.getFieldHtml(field.val()), visible: !field.edit"></div>
                         <div data-bind="jqueryElement: field.ui, visible: field.edit"></div>
                         <div data-bind="jqueryElement: format_warning,
-                                        visible: field.value && !DetailScreenConfig.field_val_re.test(field.value)"></div>
+                                        visible: field.observableVal && !DetailScreenConfig.field_val_re.test(field.observableVal)"></div>
```

Comment:
Usually when I find myself editing already-kinda-long expressions inside a data-bind, I try to pull them out into functions on the model and call those directly instead. It pushes the logic out of the template and into a js file where you can reason with it more easily.